_posts/2020-12-25-shouldExecute.md

```diff
@@ -0,0 +1,94 @@
+
+When working with databases you often have different environments such as test, development, or production. In each of these environments you might want to execute different migrations, and this can now be achieved by injecting the environment with a placeholder!
+
+Once again, let's say we have the following migrations:
+
+```
+V1__tst_migration_1.sql
+V2__dev_migration_1.sql
+V3__prd_migration_1.sql
+```
+
+Migration `V1` should only be executed in the `test` environment, `V2` in the `development` environment, and `V3` in the `production` environment. Migration `V1`'s script configuration file would need to contain the line `shouldExecute=${environment}==test`. `V2`'s will need the line `shouldExecute=${environment}==development` and `V3`'s will need the line `shouldExecute=${environment}==production`.
+
+Now, if we set the value of the `${environment}` placeholder to contain the environment we are running Flyway in, we can achieve our desired result.
+
+Running `flyway -placeholders.environment=test migrate` will only apply `V1`. Similarly `flyway -placeholders.environment=development migrate` will only apply `V2`, and `flyway -placeholders.environment=production migrate` will only apply `V3`.
+
+### The benefit of setting `shouldExecute=false`
+
+Initially it may not be obvious how hardcoding `false` for `shouldExecute` would be useful, since this just has the same effect as using [cherryPick](/documentation/configuration/parameters/cherryPick) with all of the migrations you do want to execute.
+
+However, we quite often want to defer the execution of a long running migration so that it can run, for example, overnight. Using `cherryPick` can become unwieldy when the number of migrations you do want to execute is large. `cherryPick` also requires changing how you execute Flyway, which makes it less desirable in fully automated pipelines. Let's see how you can achieve this with `shouldExecute`!
+
+For the last time, let's say we have the following migrations:
+
+```
+V1__shrt_migration_1.sql
+V2__long_migration_1.sql
+V3__shrt_migration_2.sql
+```
+
+In this example, let's also assume that we have a basic automated pipeline running `flyway migrate` on each commit to version control.
+
+Since migration `V2` takes too long to execute during a work day, we decide to execute it overnight, but don't want to delay executing `V1` and `V3`. We can still execute migrations `V1` and `V3`, but first need to create a script configuration file `V2__long_migration_1.sql.conf` for `V2` with the line `shouldExecute=false`. When we commit this to version control, migrations `V1` and `V3` will automatically be applied, ignoring `V2`. 
+
+When we reach the end of the day and decide it's time to apply `V2`, we can modify the script configuration file and remove the line containing `shouldExecute=false` since the default value is `true`. Once we commit this, migration `V2` will automatically begin execution in our pipeline.
```

Comment:
You should specify that `outOfOrder` needs to be set also.

Reply:
see latest commit